modules/repofiles/upload.go

```diff
@@ -191,6 +189,7 @@ func UploadRepoFiles(repo *models.Repository, doer *models.User, opts *UploadRep
 				// Can't clean up the store, once uploaded there they're there.
 				return cleanUpAfterFailure(&infos, t, err)
 			}
+			file.Close()
```

Comment:
Isn't this equivalent to `defer file.Close()` in line 184?

Reply:
please refer to https://github.com/go-gitea/gitea/issues/13611

to see why we must use `file.Close() ` instead of `defer file.Close()`

@kdumontnu

Reply:
This could cause an issue if `file.Close` is called twice.  
https://golang.org/pkg/os/#File.Close

I agree this is needed, as defers in a loop are usually bad news. 
The defer should be removed in favor of this.

Reply:
@garfeng I see. Didn't notice the for loop. 
@jolheiser Good call. Looks like `defer file.Close()` can be removed if `file.Close()` is added to the err catch above.